consensus/parlia/parlia.go

```diff
@@ -1137,6 +1143,27 @@ func (p *Parlia) Finalize(chain consensus.ChainHeaderReader, header *types.Heade
 			return err
 		}
 	}
+
+	if p.chainConfig.IsOnFusion(header.Number) {
+		err := p.initializeFusionContract(state, header, cx, txs, receipts, systemTxs, usedGas, false)
+		if err != nil {
+			log.Error("init fusion contract failed")
+		}
+	}
+
+	// update validators every day
+	if p.chainConfig.IsFusion(header.Number) {
+		parent := chain.GetHeaderByHash(header.ParentHash)
+		if parent == nil {
+			return errors.New("parent not found")
+		}
+		if time.Unix(int64(parent.Time), 0).Day() == time.Unix(int64(header.Time), 0).Day()-1 {
```

Comment:
The condition here need exactly cross 1 day,  but I hope we need to update if it cross 2 days or 3 days.

Change to `if time.Unix(int64(parent.Time), 0).Day() < time.Unix(int64(header.Time), 0).Day() `

Reply:
Done